### PR TITLE
Improve the flow of text in the top-level README.

### PR DESCRIPTION
**Pull request by jkoshy**

### CODE REVIEW

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
Nice rework @jkoshy ! I have a few typo related comments and a structuring suggestion.

File: README.md [hunk 1]
```diff
@@ -408,133 +465,139 @@ main =
...
+very little overhead even where the task size is very
+small, because Streamly will automatically switch between
+serial or batched execution of tasks on the same CPU depending
+on whichever is more efficient.  Please see our [concurrency
+benchmarks](https://github.com/composewell/concurrency-benchmarks) for
+more detailed performance measurements, and for a comparison with the
+`async` package.
 
-## Benchmarks
+### Performance
```
[the remark this answers is not in the document]
Comment (jkoshy) [reply README.md]: Done.

File: README.md [hunk 2]
```diff
@@ -140,16 +172,18 @@ wc file =
   & Stream.fold countAll -- IO (Int, Int, Int)
 ```
 
-This example demonstrates the simple and concise API of streamly with
-excellent modularity.  Experienced Haskellers would notice that we have
-not used bytestrings, we simply use a stream of `Word8`, simplifying the
-program.
+This example demonstrates the the excellent modularity offered by
```
[the remark this answers is not in the document]
Comment (jkoshy) [reply README.md]: Fixed.

File: README.md [hunk 3]
```diff
@@ -171,20 +205,24 @@ user    0m1.935s
...
+
+In our next example we show how the task of counting words, lines,
+and bytes could be done in parallel on multiprocessor hardware.
 
-To count words in parallel we divide the stream into chunks (arrays),
-count properties in each chunk and then add all the counts.  We use the
-same code as above except that we use an array input instead of using a
-file input.
+To count words in parallel we first divide the stream into chunks
+(arrays), do the counting within in each chunk, and then add all the
```
[the remark this answers is not in the document]
Comment (jkoshy) [reply README.md]: Done.

File: README.md [hunk 4]
```diff
@@ -193,14 +231,18 @@ countArray arr =
...
-to know whether the first element of the next chunk was a space char or
-non-space to know whether the same word is continuing to the next chunk
-or if it is a new word. `partialCounts` adds a `Bool` flag to `Counts`
-returned by `countArray` to indicate whether the first character in the
-chunk is a space.
-
-``` haskell
+
+Here the function `count` and the `Counts` data type are defined in the
+`WordCount` helper module.
```
[the remark this answers is not in the document]
Comment (jkoshy) [reply README.md]: Done.

File: README.md [hunk 5]
```diff
@@ -250,32 +297,38 @@ user    0m1.952s
...
 
-See
+Please see the file
 [WordServer.hs](https://github.com/composewell/streamly-examples/blob/master/examples/WordServer.hs)
-for full working code including the imports that we may have omitted below.
+for the complete code for this example, including the imports that we
+have omitted below.
 
-``` haskell
+```haskell
```
[the remark this answers is not in the document]
Comment (jkoshy) [reply README.md:329]: Added.

File: README.md [hunk 6]
```diff
@@ -408,133 +465,139 @@ main =
...
-If you can write a program significantly faster in some other way or
-with some other language, please let us know and we will improve.
+* [Streaming Benchmarks](https://github.com/composewell/streaming-benchmarks) | [Concurrency Benchmarks](https://github.com/composewell/concurrency-benchmarks)
+* [Functional Conf 2019 Video](https://www.youtube.com/watch?v=uzsqgdMMgtk) | [Functional Conf 2019 Slides](https://www.slideshare.net/HarendraKumar10/streamly-concurrent-data-flow-programming)
+* [Other Guides](docs/)
+* [Streamly Homepage](https://streamly.composewell.com)
+
+## Notes
+
+'Streamly' comes equipped with a very powerful set of abstractions
```
[the remark this answers is not in the document]
Comment (jkoshy) [reply README.md]: Done.

File: README.md [hunk 7]
```diff
@@ -408,133 +465,139 @@ main =
...
+highly modular abstractions for building programs while also offering
+the performance close to an equivalent (imperative) C program.
 
-These results show that streamly is the fastest effectful streaming
-implementation on almost all the measured micro benchmarks. In many cases
-it is up to 100x faster and in some cases even 1000x faster. In many
-composite operation benchmarks streamly turns out to be significantly
-faster than the Haskell lists implementation.
+Streamly offers excellent performance even for byte-at-a-time stream
+operations byte using efficient abstractions like `Unfold`s and
```
[the remark this answers is not in the document]
Comment (jkoshy) [reply README.md]: Done.

File: README.md [hunk 8]
```diff
@@ -545,20 +608,26 @@ significant way:
...
+## Licensing
+
+Streamly is an [open source](https://github.com/composewell/streamly)
+project available under a liberal [BSD-3-Clause
+license](https://github.com/composewell/streamly/blob/master/LICENSE).
+
+## Contributing to Streamly
 
-## Contributing
+An an open project we welcome contributions:
```
[the remark this answers is not in the document]
Comment (jkoshy) [reply README.md]: Done.

**Comment by harendra-kumar** [comment]
Thanks @jkoshy . It made the quality of the document much better.

**Pull request merged**